api/api_test.go

```diff
@@ -538,3 +539,36 @@ func TestCloneHttpClient(t *testing.T) {
 	})
 
 }
+
+func TestClient_HeaderRaceCondition(t *testing.T) {
+	require := require.New(t)
+
+	conf := DefaultConfig()
+	conf.Headers = map[string][]string{
+		"test-header": {"a"},
+	}
+	client, err := NewClient(conf)
+	require.NoError(err)
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+
+	go func() {
+		for i := 0; i < 2500; i++ {
+			req, err := client.newRequest("GET", "/any/path/will/do")
+			require.NoError(err)
```

Comment:
`requires` and `t.FailNow` aren't meant to be only invoked from the main test goroutine: https://golang.org/pkg/testing/#B.FailNow .

Reply:
Cool,  will update this.